### PR TITLE
Mhilliard reality checks

### PR DESCRIPTION
**Pull request by seamuncle**
I think this is as far as I can go, @thauber --its not perfect and you'll probably catch stuff I missed or feels wrong to you; but I've hit it in a bunch of ways and it feels bulletproof after a swack of debugging.

Most of it hinges on calling checkCredentials instead of the original webview request to authenticate and the creation of "credentials" over and above the simple auth token.  Deprecations have been added to pull more attention to it.  Everything else should fall under the tech plan.

### CODE REVIEW

File: Betable/Betable.m [hunk 1]
```diff
@@ -42,11 +44,18 @@
...
+// int disguied as enum used in place of double disguied as NSTimeInterval.
+typedef enum heartbeatPeriods {
+    NOW = 0,
+    HEALTHY_PERIOD = 60,
+    UNHEALTHY_PERIOD = 5
+} HeartBeatPeriods;
+
 @interface Betable () {
-    NSMutableArray *_deferredRequests;
+    NSMutableOrderedSet  *_deferredRequests;
```
Comment (thauber) [inline Betable/Betable.m:58]: This was only for loading of the profile. I don't think we need to defer any requests anymore, but I could be wrong.

Comment (seamuncle) [inline Betable/Betable.m:58]: [NSMutableArray array] was Apple-deprecated so I changed it to something else.

I couldn't find a case where it was used while I was talking to localhost; but I didn't want to assume.  That entire bit of logic can be streamlined out if you think the simplicity is worthwhile.

File: Betable/BetableHandlers.h [hunk 2]
```diff
@@ -5,8 +5,36 @@
...
\ No newline at end of file
+typedef void (^BetableCancelHandler)();
+typedef void (^BetableLogoutHandler)();
+
+// Game pust provide an implementation of this to betable to facilitate proper game event interactions
+@protocol BetableCredentialCallbacks<NSObject>
+
+@required
+// Betable needs up-to-date insight to current view controller is should game be backgrounded and foregrounded
+- (UIViewController*) currentGameView;
```
Comment (thauber) [inline Betable/BetableHandlers.h:21]: Does the app need to keep this value up to date for every transition they do?

Comment (thauber) [inline Betable/BetableHandlers.h:21]: Or is it more of just marking the rootviewcontroller

Comment (seamuncle) [inline Betable/BetableHandlers.h:21]: In practice, its probably the latter.  But I couldn't convince myself that would always be the case--not just usually, especially given Apples perchance for changing the way stuff works--if its not, then lets give the game developer some wiggle room.

As a matter of convenience, I could make the method optional and fallback to the default for UIApplication...

Comment (thauber) [inline Betable/BetableHandlers.h:21]: Nope, I think this is fine.

**Comment by thauber** [comment]
lgtm! We will want to take the deferred request stuff out at some point, but it shouldn't hurt anything atm.

**Pull request merged**